lib/puppet_x/puppetlabs/cisco_ios/utility.rb

```diff
@@ -48,7 +48,7 @@ def self.facts(facts) # rubocop:disable Style/TrivialAccessors
     end
 
     def self.ios_device_type
-      unless @facts['hardwaremodel'].nil?
+      if @facts && !@facts['hardwaremodel'].nil?
```

Comment:
We should not be changing module code at the same time as PDK updates.

Reply:
We should actually, this was a bug / issue found by running pdk update. We now run tests in paralllel, and this fix was required. It did not warrant a separate pr.